rule.py

```diff
@@ -169,35 +218,50 @@ def get_eaten(self, eater, old_pos, new_pos):
         self.board_map_2d[new_pos.y][new_pos.x] = eater 
         self.board_map_2d[old_pos.y][old_pos.x] = self._IS_EMPTY
 
+    def can_I_eat_this(self, old_pos, new_pos): 
```

Comment:
- Check the return value True for what case and False for what case
- This seems a little repetitive with the is_interfered() method. You might want to consider to change it so it is not so repetitive.